packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -31,17 +31,9 @@ export interface ISharedTreeCoreEvents extends ISharedObjectEvents {
  * TODO: actually implement
  * TODO: is history policy a detail of what indexes are used, or is there something else to it?
  */
-export class SharedTreeCore<TChangeRebaser extends ChangeRebaser<any, any, any>>
+export class SharedTreeCore<TChange, TChangeFamily extends ChangeFamily<any, TChange>>
```

Comment:
I'm guessing in the long term we'd rather SharedTreeCore just be parameterized over the change family. Addressing that would require addressing it in the EditManager. Did you ever find out what was problematic about the `infer` approach?

This comment should not block this PR. Feel free to merge without resolving it.

Reply:
Craig said it often incorrectly infers the types as `any`.